codegen/generator/go/templates/src/object.go.tmpl

```diff
@@ -11,6 +11,17 @@ type {{ .Name | FormatName }} struct {
 	{{- end }}
 }
 {{- end }}
+
+
+{{- if or (eq (.Name | FormatName) "Directory") (eq (.Name | FormatName) "Container") }}
```

Comment:
If we want to change the types it gets applied to, we just need to adjust this line.

Reply:
As explained in:
- https://github.com/dagger/dagger/issues/3885#issuecomment-1502081027

We can do a follow up to make codegen smarter on which object types to select, instead of having to manually hardcode this. I don’t think we need to wait for that to merge this.